bullet_train-outgoing_webhooks/app/jobs/webhooks/outgoing/endpoints_deactivation_job.rb

```diff
@@ -0,0 +1,17 @@
+class Webhooks::Outgoing::EndpointsDeactivationJob < ApplicationJob
+  queue_as :default
+
+  def perform
+    return unless automatic_deactivation_endpoint_enabled?
+
+    endpoint_health = Webhooks::Outgoing::EndpointHealth.new
+    endpoint_health.deactivate_failed_endpoints!
```

Comment:
First, deactivate to avoid processing the endpoints that have just been marked.

Reply:
Seems like a weird naming combo: `endpoint_health.deactivate_failed_endpoints!`. Maybe something like `Webhooks::Outgoing::EndpointDoctor` :D